ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/ImagesTableContainer.tsx

```diff
@@ -85,7 +88,10 @@ function ImagesTableContainer({
                     onWatchImage={onWatchImage}
                     onUnwatchImage={onUnwatchImage}
                     showCveDetailFields={showCveDetailFields}
-                    onClearFilters={() => setSearchFilter({})}
+                    onClearFilters={() => {
+                        onFilterChange({});
+                        pagination.setPage(1);
```

Comment:
To confirm my understanding, not to request change. Is an alternative to lift up responsibility for pagination to `WorkloadCvesOverviewPage` component:

```ts
function onFilterChange(newFilter: SearchFilter) {
    setSearchFilter(newFilter);
    pagination.setPage(1);
}
```

Therefore, its tracked change callback would become:

```tsx
            onFilterChange={(newFilter, searchPayload) => {
                onFilterChange(newFilter);
                trackAppliedFilter(WORKLOAD_CVE_FILTER_APPLIED, searchPayload);
            }}
```

And `WhateverTableContainer` would have `onFilterChange={onFilterChange}` prop?

Reply:
Yes! I do like this a lot, but was aiming to keep changes somewhat minimal. There are other instances where the `setSearchFilter` could be lifted up even further based on the same principle but I figured I'd keep the review light since I've done quite a few refactoring PRs lately. 😓

Reply:
Agree, less is more. Thank you for the steady progress with steps that are safe to do and clear to review.